### ISSUE TITLE
Feature request: Prevent routing to component

### ISSUE THREAD

**Issue by Bodeclas** [willmendesneto/ngx-feature-toggle#187]
**I'm submitting a ...** (check one with "x")

[ ] bug report => search github for a similar issue or PR before submitting
[X] feature request

**Expected behavior**

Prevent routing to component

If access to a component should be prevented when a feature is turned off. This can be accomplished by using the canActivate option available in Angular’s router.

**What is the motivation / use case for changing the behavior?**

I want to load only modules when a feature is turned on.

**Comment by willmendesneto** [comment]
A great request! I'm thinking if that can be done by extending `feature-toggle-service` and exposing an Angular guard that can be used and/or extended if required.

I'm kinda busy these days, so feel free to open a pull request with these changes, otherwise I can check that in the next weeks.

**Comment by heavybeard** [comment]
Maybe my comments is overscope of this issue, but may could help to think about it.

A look at https://github.com/AlexKhymenko/ngx-permissions package could help to implement this feature and others too

**Comment by willmendesneto** [comment]
Unfortunately, I didn't have the proper time to apply these changes in the module. I applied some of the changes into `feature-toggle-service` and updated the repository to use the latest version of the package.

For now, I encourage to use the feature toggle service before the routing solution, create a guard and do the check inside the guard.


```ts
// File with module and route definitions
...
import { set } from 'feature-toggle-service';
...
set({
  parent: true,
  children: true
});
...
// route definition
const routes: Routes = [
  { 
    ...
    data: {
      toCheck: 'parent',
    },
    ...
  }
];
```

```ts
// guard file
...
@Injectable()
export class AuthChildGuard implements CanActivateChild {

  constructor(private router: Router) {
  }

  canActivateChild(route: ActivatedRouteSnapshot, state: RouterStateSnapshot) {
    return this.check(route);
  }

  canLoad(route: Route) {
    return this.check(route);
  }

  check(route: ActivatedRouteSnapshot | Route) {
    const toCheck = !!route && !!route.data && !!route.data['toCheck'] ? route.data['toCheck'] : '';
    // returning true in case the data param was not passed via route
    return !!toCheck ? isOn(toCheck) : true;
  }
}
```

Just checked and it works and updates as expected. You can check an example in this link https://stackblitz.com/edit/ngx-feature-toggle-guard-sample

I'll try to apply the changes in the next days, but feel free to open a pull request applying these changes.

**Comment by willmendesneto** [comment]
Hi @Bodeclas this package now has 2 route guards:

- `NgxFeatureToggleCanLoadGuard` to be used in `canLoad` route configuration

```js
import { NgxFeatureToggleCanLoadGuard } from 'ngx-feature-toggle';
...
export const routes: Routes = [
  {
    path: 'home',
    component: HomeComponent,
    canLoad: [NgxFeatureToggleCanLoadGuard],
    data: {
      featureToggle: ['enableSecondText'],
    },
  },
];
...
```

- `NgxFeatureToggleCanActivateGuard ` to be used in `canActivate` route configuration

```js
import { NgxFeatureToggleCanActivateGuard } from 'ngx-feature-toggle';
...
export const routes: Routes = [
  {
    path: 'home',
    component: HomeComponent,
    canActivate: [NgxFeatureToggleCanActivateGuard],
    data: {
      featureToggle: ['enableSecondText'],
    },
  },
];
...
```

Also, it accepts multiple checks based on different feature toggles. E.G.

```js
...
export const routes: Routes = [
  {
    path: 'home',
    component: HomeComponent,
    canActivate: [NgxFeatureToggleCanActivateGuard],
    data: {
      featureToggle: [
        // This configuration will check if feature toggle is enabled
        'enableSecondText',
        // This configuration will check if feature toggle is disabled
        // since it has `!` prefix in the configuration
        '!enableFirstText'
      ],
    },
  },
];
...
```

In this case, we are combining the checks. So the component will be activated if `enableSecondText` is configured as `true` AND `enableFirstText` is configured as `false`. With that configuration you can have all the flexibility to cover different scenarios in your app.

I'm closing this issue for now since it's solved. Thanks for raising that!

**Comment by willmendesneto** [comment]
New feature published and avaibale in `ngx-feature-toggle@7.1.0`  🎉
